### ISSUE TITLE
Unfixable `no-await-expression-member` inside condition

### ISSUE THREAD

**Issue by felipecrs** [sindresorhus/eslint-plugin-unicorn#2557]
```ts
// Handle creating/removing shortcuts on Windows when installing/uninstalling.
if ((await import("electron-squirrel-startup")).default) {
  console.error("Squirrel event handled. Quitting...");
  app.exit();
}
```

This triggers [no-await-expression-member](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v56.0.1/docs/rules/no-await-expression-member.md), but this is unfixable because it's in the root of my script.

If I do `const {default: squirrelStartup} = await import("electron-squirrel-startup")`, then `squirrelStartup` will be kept declared during the whole lifecycle of my application, while I only needed it to be available for the evaluation of the conditional.

**Comment by fisker** [comment]
Please add a variable declaration outside, this is expected.

**Comment by felipecrs** [comment]
@fisker, have you read this part?



> If I do `const {default: squirrelStartup} = await import("electron-squirrel-startup")`, then `squirrelStartup` will be kept declared during the whole lifecycle of my application, while I only needed it to be available for the evaluation of the conditional.

**Comment by fisker** [comment]
What do you mean, 

> will be kept declared during the whole lifecycle of my application

It will be `gc`ed.

---

You can always add a block, if you don't want the variable accessible in other places.

```js
{
  const {default: squirrelStartup} = await import("electron-squirrel-startup")
  // if (...) 
}
```

**Comment by felipecrs** [comment]
> It will be `gc`ed.

As far as I understand, constants are not garbage collected in JavaScript except when their context `{}` ends. Am I wrong?

> You can always add a block, if you don't want the variable accessible in other places.

lol. I had forgotten about that. Thank you.

**Comment by fisker** [comment]
> constants are not garbage collected in JavaScript except when their context {} ends. 

I'm not sure about it, I think it can be "gc"ed inside `if`,  since there is no reference to it anymore, but I don't have proof.
